File: Sexta_Generacion_CPCFI/clases/dia12/1634.cpp
// cses
#include <bits/stdc++.h>
using namespace std;
#define INF 1000000000
typedef long long ll;

int main() {
    int n, x = 0;
    cin >> n >> x;
    vector<ll> ci(n);
    for(int i = 0; i < n; i++) {
        cin >> ci[i];
    }
    
    vector<ll> dp(x+1, INF);
    dp[0] = 0;
    
    for(int i = 1; i <= x; i++) {
        for(int j = 0; j < n; j++) {
            if(ci[j] > i || dp[i - ci[j]] == INF) continue;
            dp[i] = min(dp[i], dp[i - ci[j]] + 1);
        }
    }
    
    if(dp[x] != INF) cout << dp[x] << endl;
    else cout << -1 << endl;
}

File: Sexta_Generacion_CPCFI/clases/dia12/1634.py
# cses
def solve(n,x,ci):
    dp = [float('inf')] * (x+1)
    dp[0] = 0
    
    for i in range(1, x+1):
        for j in range(n):
            if (ci[j] > i) or  (dp[i-ci[j]] == float('inf')):
                continue
            dp[i] = min(dp[i], dp[i-ci[j]] + 1)
    
    if dp[x] != float('inf'):
        return dp[x]
    else:
        return -1

n, x = map(int, input().split())
ci = list(map(int, input().split()))
print(solve(n,x,ci))
